DependencyInjection/Configuration.php

```diff
@@ -83,6 +83,8 @@ public function getConfigTreeBuilder()
                             ->defaultValue('auto')
                         ->end()
                         ->scalarNode('formatter')->defaultNull()->end()
+                        ->booleanNode('inject_debugger_for_discovered_http_client')->defaultTrue()->end()
```

Comment:
> Correct me if I am wrong, but this will register the default client in the discovery, will it not?

No, It will find take a (new) unconfigured client and add toolbar debugging on that and then make sure that this client is found by discovery in the future. 

This PR also lets to you decide if you want to use a configured client as auto discovery. 

> Can we find a shorter name? Like profile_discovered_client and profile_discovered_async_client

Yes please. It was quite late last night when I wrote this. =)